### DISCUSSION TITLE
Possible to have "stacked" launch menu's?

### DISCUSSION THREAD

**Discussion by zonywhoop** [wezterm/wezterm#3709] [General]
I love the launch menu concept but really want to stack or make it multi layer system where I can put customer / work terminals under a menu, personal servers under another, etc.  Is there a way to solve this currently or would it be a new feature?

**Comment by wez** [comment]
Take a look at: https://wezfurlong.org/wezterm/config/lua/keyassignment/InputSelector.html

**Reply by zonywhoop** [reply]
So this sort of works, but is it possible to have inputSelectors with cascading inputSelectors?  e.g. I have a project named Acme and when I choose that I want to have options for say vscode, terraform, ansible, where it sets up the venvs, etc in the shell for working on that project.  Been playing with it off and on and just haven't figured out the menu to menu scenario yet

**Reply by wez** [reply]
In your action_callback from the first one, you can [window:perform_action](https://wezfurlong.org/wezterm/config/lua/window/perform_action.html) another InputSelector for the next stage, and so on

**Reply by danjessen** [reply]
> In your action_callback from the first one, you can [window:perform_action](https://wezfurlong.org/wezterm/config/lua/window/perform_action.html) another InputSelector for the next stage, and so on

Would this also be a way / the way to reload the list? 

Coming from fzf where it is possible to have previews, keybinds, reloads etc. Even the possibility to style and change the pop up. The InputSelector is great. But more flexibility would make it even better.

Basic previewing could maybe be leveraged with bat from sharkdp. It is another rust project after all.
